### PR TITLE
Terraform Plan output variable to detect changes

### PR DESCRIPTION
**Pull request by wradstok**
This work was done by BlueBasher in #856, but went unnoticed and could no longer be automatically merged. I merely updated the version numbers and applied it to V2 as well,

### CODE REVIEW

**Comment by mericstam** [comment]
Just got back from x-mas break. I will test the added code and get back to you.

File: Extensions/Terraform/Src/Tasks/TerraformTask/TerraformTaskV2/src/base-terraform-command-handler.ts [hunk 1]
```diff
@@ -204,6 +204,7 @@ export abstract class BaseTerraformCommandHandler {
 
     public async plan(): Promise<number> {
         await this.onlyPlan();
+        tasks.setVariable('changesPresent', exitCode === 2);
```
Comment (JaydenMaalouf) [inline Extensions/Terraform/Src/Tasks/TerraformTask/TerraformTaskV2/src/base-terraform-command-handler.ts:207]: does exitCode need to be defined here?
Something like
`let exitCode = await this.onlyPlan();`

**Comment by mericstam** [comment]
Hi, The reason why I am holding back this PR a little is because  we are in the process of moving the terraform extension to a separate repository.

**Comment by JaydenMaalouf** [comment]
> Hi, The reason why I am holding back this PR a little is because we are in the process of moving the terraform extension to a separate repository.

Let me know if you'd like a PR raised in the new repository once it's migrated for this change.

**Comment by mericstam** [comment]
I got sick in covid. so everything is a bit delayed. Planning for move and new release this week.. I will notify in this conversation when you can fork create the new PR in new location

**Comment by JaydenMaalouf** [comment]
> I got sick in covid. so everything is a bit delayed. Planning for move and new release this week.. I will notify in this conversation when you can fork create the new PR in new location

Thanks! Appreciate the follow up. Hope you're feeling better and if not, get well soon!

**Comment by mericstam** [comment]
Hi new repo location:  https://github.com/microsoft/azure-pipelines-terraform
Please fork create a new PR in the new location and we can continue with this fix.

**Comment by JaydenMaalouf** [comment]
> Hi new repo location: https://github.com/microsoft/azure-pipelines-terraform Please fork and create a new PR in the new location and we can continue with this fix.

Amazing!
Here is the PR:
https://github.com/microsoft/azure-pipelines-terraform/pull/11

**Comment by mericstam** [comment]
closing this PR since it has moved to https://github.com/microsoft/azure-pipelines-terraform/pull/11

**Pull request closed without merging**